V2.19.1

ELundby45 opened the pull request:
## Notes:
- win-64 and osx-arm64 can be built on CI, but the graph had to be manually created to extend timeout.
- linux-64: Can build on CI if CPU_COUNT is set to 1, but the build then takes 18 hours. 
- linux-aarch64: Couldn't find a way to get past out of memory errors despite limiting all parallelism and memory boxing bazel. 

## Changes:

- Re-vendor protobuf
    - Disable patches specifically used to un-vendor
    - Remove actions in build.sh related to un-vendoring
    - Remove it from TF_SYSTEM_LIBS
- Remove libtensorflow and libtensorflow_cc to match upstream. 
    - Remove from meta.yaml
    - Cleanup build script and remove everything related to libtensorflow*
- Remove toco entry points as they have been removed upstream (deprecated in 2.16.1)
- Re-based existing patches that are still needed
- Added new patches
    -  0033-fix-osx-wheel-name.patch: Fix the generated wheels name on OSX
    -  0034-fix-convolution-thunk-countdown-signature.patch Pull in upstream XLA fix
    - 0035-remove-ambiguous-defaultquantparamspass-constructor.patch
    - 0036-fix-win-binary-quirk.patch Fixes a unixy cmd call on windows
    - 0037-explicitly-link-in-_pywrap_profiler_plugin.patch Fix for missing symbols on linux-aarch64 that weren't being found via lazy binding. 

## Windows Specific:
- move builds to the root drive ( ie "C:\b-o" ) to eliminate long path errors. 
- disable avx512vnni, same as unix

## Links
- osx-arm64 and win-64 can be found here:  https://package-build.anaconda.com/v1/graph/e55ec071-3f7a-4aa0-9e4b-33cc1f8ba6e2
- linux-aarch64, linux-64 and linux-64-cuda artifacts can be found here: https://anaconda.org/build/repo
    - logs attached
    
[tensor_linux-64_cpu.log](https://github.com/user-attachments/files/22702494/tensor_linux-64_cpu.log)
[tensor_linux-64_cuda.log](https://github.com/user-attachments/files/22702513/tensor_linux-64_cuda.log)
[tensor_linux-aarch64.log](https://github.com/user-attachments/files/22702467/tensor_linux-aarch64.log)

ELundby45 committed:
v2.19.0

ELundby45 committed:
rebase patches.

ELundby45 committed:
Add osx specific patch.

ELundby45 committed:
Merge branch 'v2.18.1-win' into v2.19.0

ELundby45 committed:
Stashing patch.

ELundby45 committed:
Update 0018-add-absl_string_view-target.patch

ELundby45 committed:
Update from main

ELundby45 committed:
2.19.1

ELundby45 committed:
pbp workaround attempts.

ELundby45 committed:
linux-* memory optimization.

ELundby45 committed:
mv host_jvm_args to startup opts.

ELundby45 committed:
Further limit resources.

ELundby45 committed:
Remove libtensorflow_cc/libtensorflow

ELundby45 committed:
Revert linux memory optimizations.

ELundby45 committed:
fixes linux-aarch64.

ELundby45 committed:
Remove unnecessary changes.

ELundby45 committed:
Rename patch add description

ELundby45 committed:
Try shutting down before clean

Update bld.bat

JamesRobertsonGames commented on recipe/meta.yaml:
```diff
@@ -242,21 +236,21 @@ outputs:
         # github.com/tensorflow/tensorflow/blob/v{{ version }}/tensorflow/tools/pip_package/setup.py
         - absl-py >=1.0.0
         - astunparse >=1.6.0
+        - python-flatbuffers >=24.3.25
```
Flatbuffers is pinned to 24.3.25 - should this not stay inline?

ELundby45 replied:
I was following upstream:
https://github.com/tensorflow/tensorflow/blob/v2.18.1/tensorflow/tools/pip_package/setup.py#L81C5-L81C30

But, I think you are right after looking through the code. Lots of header files lock to the specific version:
https://github.com/tensorflow/tensorflow/blob/v2.18.1/tensorflow/compiler/mlir/lite/schema/conversion_metadata_generated.h#L25-L29

ELundby45 committed:
Add strict pinning for flatbuffers.

See upstream:

static_assert(FLATBUFFERS_VERSION_MAJOR == 24 &&
              FLATBUFFERS_VERSION_MINOR == 3 &&
              FLATBUFFERS_VERSION_REVISION == 25,
             "Non-compatible flatbuffers version included");

JamesRobertsonGames reviewed (approved):
🥳
